packages/react-components/src/components/MessageThread.tsx

```diff
@@ -849,7 +849,7 @@ export const MessageThread = (props: MessageThreadProps): JSX.Element => {
       if (onLoadPreviousChatMessages) {
         isLoadingChatMessagesRef.current = true;
         // Fetch message until scrollTop reach the threshold for fetching new message
-        while (!isAllChatMessagesLoadedRef.current && chatScrollDivRef.current.scrollTop <= 500) {
+        while (!isAllChatMessagesLoadedRef.current && chatScrollDivRef.current.scrollTop <= 0) {
```

Comment:
The logic here is for smoothing the infinite scroll - the chat thread will be starting pulling your messages before you actually hit the top of the message (pulling only when scrollTop equals 0 will make it feel super laggy when scrolling up)

So this will actually fix the problem, but it changes the experience of infinite scrolling

Reply:
I just looked around and found out Safari doesn't support overflow-anchor propery - so there is no way from css side to change its behavior of doing this "sticky on the top" problem.

But we do have the full control from js side - https://kirbysayshi.com/2013/08/19/maintaining-scroll-position-knockoutjs-list.html this sounds like a promising solution with [JSFiddle](https://jsfiddle.net/kirbysayshi/57HbV/)

Maybe we can enable this polyfill when it is under Safari

Reply:
@PorterNan ,
Good point. I figured that it was for a "smooth" experience.
However, I did notice that even on chrome, if you drag the scrollbar high enough it can also get stuck and keep scrolling. It's less obvious compared to on Safari, but it is reproducible. 

What do you think would be a good alternative to this?

Reply:
If you look at the code around line 871 - 891, it is targeting the problem for chrome, maybe you can change the threshold a bit (set it as 50px or something more, if 50px is not working try something like percentage of totally height) then test if the bug would be fixed in Safari (also check chrome if you can repro it well)